### PR TITLE
Cleanup header builder

### PR DESCRIPTION
**Pull request by jrudolph**
A first step of clean-ups in HeaderBuilder / EnvelopeBuffer

### CODE REVIEW

**Comment by drewhk** [comment]
LGTM.

**Comment by ktoso** [comment]
LGTM, feel free to merge after validation.

**Comment by jrudolph** [comment]
Great, will do! Thanks for the quick review.

File: akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala [hunk 1]
```diff
@@ -71,6 +71,29 @@ private[akka] class DirectByteBufferPool(defaultBufferSize: Int, maxPoolEntries:
       tryCleanDirectByteBuffer(buf)
   }
 
+  private final def tryCleanDirectByteBuffer(toBeDestroyed: ByteBuffer): Unit = DirectByteBufferPool.tryCleanDirectByteBuffer(toBeDestroyed)
+}
+
+/** INTERNAL API */
+private[akka] object DirectByteBufferPool {
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala]: could we even move this to ` akka.io.DirectByteBufferPool`, which has a similar thing?

**Comment by patriknw** [comment]
LGTM, with one question

File: akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala (hunk 1 above)
Comment (jrudolph) [inline akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala]: That's exactly the file ;)

Comment (jrudolph) [inline akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala]: But it's easy to confuse, copy'n paste made this occurrence look almost like the other one.

Comment (patriknw) [inline akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala]: LOL, missed that

**Comment by patriknw** [comment]
LGTM

**Pull request merged**